libcontainer/cgroups/fs2/memory.go

```diff
@@ -32,13 +32,21 @@ func numToStr(value int64) (ret string) {
 }
 
 func setMemory(dirPath string, cgroup *configs.Cgroup) error {
-	if val := numToStr(cgroup.Resources.MemorySwap); val != "" {
-		if err := fscommon.WriteFile(dirPath, "memory.swap.max", val); err != nil {
+	mem := numToStr(cgroup.Resources.Memory)
+	memSwap := numToStr(cgroup.Resources.MemorySwap)
+	if mem == "max" {
+		// If the memory is set to max, set the mem+swap to max, too
+		memSwap = "max"
+	}
+
+	if memSwap != "" {
+		if err := fscommon.WriteFile(dirPath, "memory.swap.max", memSwap); err != nil {
```

Comment:
to keep the same cgroup v1 behaviour (as the config.json file is configured for), we need to set the memory swap to `requested_memory_limit - requested_swap_limit`.  Since on cgroup v1 the swap limit accounts also for the memory

Reply:
Hmm, in this case maybe we don't need to touch swap at all, and this PR should be closed, right?

Reply:
yes setting the `memory.max` should be enough on cgroup v2 as the swap max should not have any effect on it.

Reply:
but since the PR is open, I think we could use it for setting the swap limit as it is expected to work on cgroup v1

Reply:
See #2288